EIPS/eip-draft_request_id.md

```diff
@@ -0,0 +1,140 @@
+
+**Q: The efficency gains might encourage clients to flood their peers with too many simultanous requests**
+
+Peers can always throttle or disconnect if they don't feel treated well. This is the same as today.
+
+**Q: If `les` already defines the commands like this, why not just use the `les` protocol?**
+
+In practice, peers that serve the `les` protocol are much harder to find in the network. The reasons for this are varied but might boil down to client defaults, immature implementations or missing incentives.
+
+**Q: Networking works today, isn't this just adding bloat?**
+
+This is adding a single integer per command while at the same time reducing code complexity and improving networking effiency. The addition seems justified.
+
+## Backwards Compatibility
+<!--All EIPs that introduce backwards incompatibilities must include a section describing these incompatibilities and their severity. The EIP must explain how the author proposes to deal with these incompatibilities. EIP submissions without a sufficient backwards compatibility treatise may be rejected outright.-->
+This EIP extends the `eth` protocol in a backwards incompatible way and requires rolling out a new version, `eth/65`. However, `devp2p` supports running multiple versions of the same wire protocol side-by-side, so rolling out `eth/65` does not require client coordination, since non-updated clients can keep using `eth/64`.
+
+This EIP does not change the consensus engine, thus does *not* require a hard fork.
+
+## Test Cases
+<!--Test cases for an implementation are mandatory for EIPs that are affecting consensus changes. Other EIPs can choose to include links to test cases if applicable.-->
+
+TODO
+
+## Implementation
+<!--The implementations must be completed before any EIP is given status "Final", but it need not be completed before the EIP is accepted. While there is merit to the approach of reaching consensus on the specification and rationale before writing code, the principle of "rough consensus and running code" is still useful when it comes to resolving many discussions of API details.-->
+
+TODO
+
+## Security Considerations
+<!--All EIPs must contain a section that discusses the security implications/considerations relevant to the proposed change. Include information that might be important for security discussions, surfaces risks and can be used throughout the life cycle of the proposal. E.g. include security-relevant design decisions, concerns, important discussions, implementation-specific guidance and pitfalls, an outline of threats and risks and how they are being addressed. EIP submissions missing the "Security Considerations" section will be rejected. An EIP cannot proceed to status "Final" without a Security Considerations discussion deemed sufficient by the reviewers.-->
+
+None?
```

Comment:
hehe :laughing:

I'm also drawing a blank.

We're introducing a predictable part of the message, making known-plaintext attacks easier? ECIES (any encryption, really) should be resistant to that though.

Reply:
Just to make sure I understand the known-plaintext attack scenario. We are talking about the risk of reduced security for encrypted parts of the contents such as a block signer in Görli?

Because, afaik, the commands in itself aren't encrypted (just compressed) anyway.

---

I mean, the template says any EIP without this sections is rejected so I probably just leave it in and see if someone points out something important during the process.

Reply:
Ah wait, that's not right. The current communication *is* in fact already encrypted.

Reply:
Yeah, you should probably just ignore me here, I brought it up because it's the only thing I could think of, but I don't think this makes the encryption meaningfully easier to break.